lightly_studio/src/lightly_studio/utils/download.py

```diff
@@ -0,0 +1,82 @@
+"""Module for downloading example datasets from the web."""
+
+import os
+import shutil
+import zipfile
+from pathlib import Path
+
+import requests
+from tqdm import tqdm
+
+# URL to download the main branch of the repo as a zip
+ZIP_URL = "https://github.com/lightly-ai/dataset_examples/archive/refs/heads/main.zip"
+# name of the folder inside the zip
+REPO_DIR_IN_ZIP = "dataset_examples-main"
+
+
+def download_example_dataset(target_dir: str = "dataset_examples", force: bool = False) -> str:
+    """Downloads the lightly-ai/dataset_examples repository from GitHub.
+
+    Args:
+        target_dir:
+            The directory where the dataset will be saved.
+        force:
+            If True, will download and overwrite existing data.
+            If False, will skip download if target_dir exists.
+
+    Returns:
+        The path to the downloaded dataset directory.
+    """
+    # Normalize paths
+    target_path = Path(target_dir).expanduser().resolve()
+    zip_path = target_path.with_name(f"{target_path.name}.zip")
+    temp_extract_dir = target_path.with_name(f"{target_path.name}_temp_extract")
+
+    # Check if data already exists
+    if target_path.exists():
+        if not force:
+            print(
+                f"'{target_path}' already exists. Skipping download. Use force=True to re-download."
+            )
+            return str(target_path)
+        print(f"'{target_path}' exists. Forcing re-download...")
+        shutil.rmtree(path=target_path)
```

Comment:
ChatGPT5-code-high:

This deletes the user’s current dataset as soon as force=True, before a new archive has been downloaded or extracted. Any network failure after that point leaves the user with no dataset at all. Better download/extract into a temporary location first (as it is already done) and only replace the existing directory once the new data is verified. I.e. do these two lines directly after each other

```python
shutil.rmtree(path=target_path)
shutil.move(src=str(temp_extract_dir / REPO_DIR_IN_ZIP), dst=str(target_path))
```